.github/workflows/bot.yml

```diff
@@ -156,3 +156,104 @@ jobs:
         uses: stefanzweifel/git-auto-commit-action@v5
         with:
           commit_message: "style: apply lint fixes"
+
+  Criterion:
+    name: Criterion Benchmarks
+    if: contains(github.event.pull_request.labels.*.name, 'benchmark')
+    runs-on: benchmarking-runner
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install Critcmp
+        run: cargo install critcmp
+
+      - name: Determine Base Branch Commit
+        id: determine_base_commit
+        run: echo "::set-output name=base_commit::$(git rev-parse ${{ github.event.before }})"
+
+      - name: Run Criterion on branch
+        run: |
+          cargo bench --features criterion -- --save-baseline new_branch
+
+      - name: Switch to main
+        run: |
+          git fetch
+          git checkout ${{ steps.determine_base_commit.outputs.base_commit }}
+
+      - name: Run Criterion on main
+        run: |
+          cargo bench --features criterion -- --save-baseline main_branch
+
+      - name: Execute criterion_benchmark.sh
+        run: |
+          echo -n > benches/critcmp.txt && > table_output.txt
+          bash .github/scripts/criterion_benchmark.sh > table_output.txt && cat table_output.txt
+
+      - id: get_comment_body
+        name: Get Comment Body
+        run: |
+          body="$(cat table_output.txt)"
+          delimiter="$(openssl rand -hex 8)"
+          echo "body<<$delimiter" >> $GITHUB_OUTPUT
+          echo "$body" >> $GITHUB_OUTPUT     
+          echo "$delimiter" >> $GITHUB_OUTPUT
+
+      - name: Create Commit Comment
+        uses: peter-evans/commit-comment@v3
+        with:
+          sha: ${{github.event.pull_request.head.sha}}
+          body: ${{steps.get_comment_body.outputs.body}}
+
+  Iai_callgrind:
+    name: iai_callgrind_benchmarks
+    if: contains(github.event.pull_request.labels.*.name, 'benchmark')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install iai-callgrind-runner
+        run: cargo install --version 0.7.3 iai-callgrind-runner
```

Comment:
new version of iai-callgrind is out see https://github.com/iai-callgrind/iai-callgrind/blob/main/CHANGELOG.md .

There is now option to fail on bench regression and support for machine output, so probably this pr could be simplified with its update